Add stratis-min tests

bgurney-rh opened the pull request:
Related https://github.com/stratis-storage/stratisd/issues/3299

mulkieran commented on tests/stratis_min.rs:
```diff
@@ -214,6 +214,22 @@ fn test_stratis_min_pool_rename() {
+#[test]
+// Test stopping a pool using an invalid name, that will fail with a
+// "Uuid error".
+fn stratis_min_pool_stop_invalid_name() {
```
As written, the test is almost a parser test because the stratis-min code is interpreting the name as a UUID, trying to read it in, and failing, hence the UUID error. Since it never interacts with stratisd-min at all, you should use just the one part structure, like ```test_stratis_min_report_bad_subreport``` does.

bgurney-rh replied:
OK; I think I see what's going on.  I split the two tests; one stops a started pool, and the other is the parser test.

mulkieran commented on tests/stratis_min.rs:
```diff
@@ -214,6 +214,63 @@ fn test_stratis_min_pool_rename() {
+#[test]
+// Test stopping a pool using an invalid name, that will fail with a
+// "Uuid error".
+fn test_stratis_min_pool_stop_invalid_name() {
```
Best to move this one test above the "tests with sim engine" comment. Otherwise, all the tests look good as far as I can tell.

bgurney-rh replied:
OK; resolved.

bgurney-rh commented:
Removed a couple of tests (test_stratis_min_init_cache_add_data_add_cache and test_stratis_min_create_pool_encrypted_rebind); also removed the ".gitignore" changes.

bgurney-rh commented:
Squashed into a single commit.
